app/services/ResponsiblePeopleService.scala

```diff
@@ -38,9 +39,9 @@ class ResponsiblePeopleService @Inject()(val dataCacheConnector: DataCacheConnec
   def updateFitAndProperFlag(responsiblePeople: Seq[ResponsiblePerson], indices: Set[Int]): Seq[ResponsiblePerson] =
     responsiblePeople.zipWithIndex.map { case (rp, index) =>
       val updated = if (indices contains index) {
-        rp.hasAlreadyPassedFitAndProper(Some(true))
+        rp.approvalFlags(ApprovalFlags(hasAlreadyPassedFitAndProper = Some(true)))
```

Comment:
Could this not just be 

val updated =  rp.approvalFlags(ApprovalFlags(hasAlreadyPassedFitAndProper = Some(indices contains index)))